nsxt/resource_nsxt_policy_ip_address_allocation.go

```diff
@@ -125,6 +125,37 @@ func resourceNsxtPolicyIPAddressAllocationCreate(d *schema.ResourceData, m inter
 		return handleCreateError("IPAddressAllocation", id, err)
 	}
 
+	if allocationIP == "" {
+		// Wait for actual IP allocation
```

Comment:
Would be nicer to arrange this in a func. Otherwise LGTM.

Reply:
I could either create it as nested function (which is less readable in my opinion) or pass a lot of parameters (d, m, client, connector, etc..)

Reply:
I agree about nested func - it has no advantage here. I thought that putting this long `if` in a func is more readable, but as you prefer, it's not really important.

Reply:
I think it's ok to not use a function for this logic, I don't see a potential use case for code re-use, the reason would be readability only.

And this code change, after all, is pretty much moving the realization check from read to create.